rspace/src/main/scala/coop/rchain/rspace/IStore.scala

```diff
@@ -51,18 +53,21 @@ trait IStore[C, P, A, K] {
 
   private[rspace] def removeAll(txn: T, channels: Seq[C]): Unit
 
-  // compare to store.joinMap.addBinding
   private[rspace] def addJoin(txn: T, channel: C, channels: Seq[C]): Unit
 
-  // compare to store.joinMap.get(c).toList.flatten
   private[rspace] def getJoin(txn: T, channel: C): Seq[Seq[C]]
 
-  // compare to store.joinMap.removeBinding
   private[rspace] def removeJoin(txn: T, channel: C, channels: Seq[C]): Unit
 
-  // compare to store.joinMap.remove
   private[rspace] def removeAllJoins(txn: T, channel: C): Unit
 
+  private[rspace] def putTrie(txn: T,
+                              key: Blake2b256Hash,
+                              value: Trie[Blake2b256Hash, GNAT[C, P, A, K]]): Unit
```

Comment:
I was wondering how returning `Unit` could be avoided here. One thought involved the `State` monad, but because of the underlying LMDB this might not make much sense.

Reply:
You're right - `State` isn't really a good fit here.  

The method returns a `Boolean`, and we could return that.  Right now we've been all YOLO and assuming it to be `true`, but I think it would be better to be disciplined and check the results. 

If we choose that route, we should refactor the other `put`s in a similar fashion.